### PR TITLE
[bitnami/redis] Support `persistentVolumeClaimRetentionPolicy` for redis

### LINKED ISSUE #19683: [bitnami/redis] Add `persistentVolumeClaimRetentionPolicy` support for Redis statefulset
### Name and Version

bitnami/redis

### What is the problem this feature will solve?

Make use of `persistentVolumeClaimRetentionPolicy` to Delete or Retain the PVCs/PVs whenever the statefulset gets deleted or scaled down.

### What is the feature you are proposing to solve the problem?

Enable helm values to support `persistentVolumeClaimRetentionPolicy` and pass it to the templates as needed.

### What alternatives have you considered?

Similarly to Redis-cluster, we would also need this feature ([PR](https://github.com/bitnami/charts/commit/e5ac69de7323048732c47928702e8eb243eff8ef)) in regular Redis as well, not only in Redis-cluster.

### PR DESCRIPTION
**Pull request by Krisztian-Molnar**
### Description of the change

Supports PVC retention policy for Redis.

### Benefits

Make use of `persistentVolumeClaimRetentionPolicy` to Delete or Retain the PVCs/PVs whenever the statefulset gets deleted or scaled down.

### Possible drawbacks

Nil

### Applicable issues

- fixes #19683

### Additional information

Nil

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by Krisztian-Molnar** [comment]
@carrodher Could you please help me in generating Readme from values.yaml? I cloned that [repo](https://github.com/bitnami-labs/readme-generator-for-helm), but it's throwing errors no matter what I do

**Comment by Krisztian-Molnar** [comment]
@fevisera Any updates on this?

File: bitnami/redis/values.yaml [hunk 1]
```diff
@@ -488,6 +488,15 @@ master:
     ## NOTE: requires master.persistence.enabled: true
     ##
     existingClaim: ""
+  ## persistentVolumeClaimRetentionPolicy
```
Comment (fevisera) [inline bitnami/redis/values.yaml:491]: Please ensure that the formatting is maintained. Each parameter should include a comment with "##" preceding it. You can refer to an example in these lines: [link](https://github.com/bitnami/charts/pull/19689/files#diff-5063c6dcbc478573073c2f1b280610c29b86379ce92e6dc3ce04b51eeb0c8fd6R516-R518).

File: bitnami/redis/values.yaml [hunk 2]
```diff
@@ -921,6 +930,15 @@ replica:
     ## NOTE: requires replica.persistence.enabled: true
     ##
     existingClaim: ""
+  ## persistentVolumeClaimRetentionPolicy
```
Comment (fevisera) [inline bitnami/redis/values.yaml:933]: Please apply the formatting correction to this section as well.

File: bitnami/redis/values.yaml [hunk 3]
```diff
@@ -1219,6 +1237,15 @@ sentinel:
     ## @param sentinel.persistence.sizeLimit Set this to enable a size limit for `emptyDir` volumes.
     ##
     sizeLimit: ""
+  ## persistentVolumeClaimRetentionPolicy
```
Comment (fevisera) [inline bitnami/redis/values.yaml:1240]: Please apply the formatting correction to this section as well.

**Comment by fevisera** [comment]
Hi @Krisztian-Molnar,

Sorry for the delay. Kindly address the formatting comments I have left.

Thanks.

File: bitnami/redis/values.yaml (hunk 1 above)
Comment (Krisztian-Molnar) [reply bitnami/redis/values.yaml:491]: Sure, done in [this commit](https://github.com/bitnami/charts/pull/19689/commits/ed5f8c72306b878bdfa0cb57251fbbd79ebdff61).

File: bitnami/redis/values.yaml (hunk 2 above)
Comment (Krisztian-Molnar) [reply bitnami/redis/values.yaml:934]: @fevisera Formatting applied.

File: bitnami/redis/values.yaml (hunk 3 above)
Comment (Krisztian-Molnar) [reply bitnami/redis/values.yaml:1242]: Formatting applied.

**Comment by Krisztian-Molnar** [comment]
> Hi @Krisztian-Molnar,
> Sorry for the delay. Kindly address the formatting comments I have left.
> Thanks.
@fevisera Sure, changes made. I accidentally committed without signing, but I fixed it, should be OK now.

**Review by fevisera — APPROVED** [review]
Thank you for your contribution and for implementing those changes!
LGTM!

**Pull request merged**